### ISSUE TITLE
Improve the various Alignment controls props handling

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#63649]
### Description

Splitting this out from https://github.com/WordPress/gutenberg/pull/63486#issuecomment-2232760002

Applies to:
- AlignmentControl > [AlignmentUI](https://github.com/WordPress/gutenberg/blob/d1211ff7de85c66d45ad144f8c80323d875f964e/packages/block-editor/src/components/alignment-control/ui.js)
- BlockAlignmentControl > [BlockAlignmentUI](https://github.com/WordPress/gutenberg/blob/d1211ff7de85c66d45ad144f8c80323d875f964e/packages/block-editor/src/components/block-alignment-control/ui.js)
- BlockVerticalAlignmentControl > [BlockVerticalAlignmentUI](https://github.com/WordPress/gutenberg/blob/d1211ff7de85c66d45ad144f8c80323d875f964e/packages/block-editor/src/components/block-vertical-alignment-control/ui.js)

These three components are very similar but they handle their props differently. which seems to introduce some confusion about how the label and description props are expected to be used. Also, these props aren't documented in the components readme and other props aren't documented as well.

AlignmentControl
It allows to pass both the `label` and `describedBy` props, to change the default values.

BlockAlignmentUI
Both the label and description are hardcoded. The values are `Align / Change alignment`, which seems redundant.
Also, the label `Align` doesn't communicate _what_ will be aligned. It's inconsistent with the AlignmentControl which does clarify `Align text`. As such, it should be `Align block`.

BlockVerticalAlignmentUI
Used for xample in the Media & Text block, where the label `Change vertical alignment` is hardcoded and there's no description. The label is inconsistent wiht `Align` and `Align text`. It should be `Align vertically`.

## Misleading descriptions
I'd think that if these controls would have clear and meaningful labels, they wouldn't need a visually hidden description in the first place. The lack of documentation and the fact the description is visually hidden leads to incorrect implementations. For example, in the Table block the alignment control [passes a label prop](https://github.com/WordPress/gutenberg/blob/d1211ff7de85c66d45ad144f8c80323d875f964e/packages/block-library/src/table/edit.js#L458) with value `Change column alignment`. That makes sense. Unfortunately, the `describedBy` is still the default one: `Change text alignment`. Honestly, it's sad to see how the development process in this project often misses to check and test anything that is 'non visual'. The consequence is that the alignment button in the Table block has:
- a label that says `Change column alignment`
- a description that says `Change text alignment`

Screen readers will announce both the label and description, thus providing a very confusing information to usdrs. Screenshots:

![Screenshot 2024-07-17 at 09 36 48](https://github.com/user-attachments/assets/54714d7c-f9b6-405c-98e9-207f3349e4bc)

![Screenshot 2024-07-17 at 09 31 31](https://github.com/user-attachments/assets/b4fb52e1-13b9-4525-8258-0eef858a8f1f)

### Step-by-step reproduction instructions

- Select a paragraph block.
- Inspect the `Align text` button in the block toolbar.
- Observe the button has an `aria-describedby` attribute that points to an element with the description `Change text alignment`. This description is a bit redundant.
- Select a Table block.
- Inspect the `Change column alignment` button in the block toolbar.
- Observe the button has an `aria-describedby` attribute that points to an element with the description `Change text alignment`. This description is a misleading.
- Select a Group block.
- Inspect the `Align` button in the block toolbar.
- Observe the button has an `aria-describedby` attribute that points to an element with the description `Change alignment`. Both the label and the description don't clarify _what_ to align. The description `Change alignment` is redundant.
- Select a Media & Text block.
- Inspect the `Change vertical alignment` button in the block toolbar.
- Observe the button does not have an `aria-describedby` attribute.
- Observe the label `Change vertical alignment` is inconsistent with the other labels, to be more consistent it should say 'Align vertically'.

### Screenshots, screen recording, code snippet

AlignmentControl in a Paragraph:

![align text](https://github.com/user-attachments/assets/8cba5016-2ea0-42c4-ab3e-b57f367eaa6e)

AlignmentControl in a Table:

![table ealign column](https://github.com/user-attachments/assets/1cb973f9-2137-45b3-8a1e-fd9b9e6b16e2)

BlockAlignmentUI in a Group:

![align block](https://github.com/user-attachments/assets/02a3fafa-c86b-4742-bdd2-4007a89eafea)

BlockVerticalAlignmentUI in a Media & Text:

![Screenshot 2024-07-17 at 12 36 38](https://github.com/user-attachments/assets/6c2ede6c-fd02-4333-9063-2da2dd536f8d)

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes
